src/Reader/IterableDataReader.php

```diff
@@ -65,50 +71,9 @@ private function sortItems(iterable $items, Sort $sort): iterable
 
     protected function matchFilter(array $item, array $filter): bool
     {
-        $operation = array_shift($filter);
-        $arguments = $filter;
-
-        switch ($operation) {
-            case Not::getOperator():
-                return !$this->matchFilter($item, $arguments[0]);
-            case Any::getOperator():
-                foreach ($arguments[0] as $subFilter) {
-                    if ($this->matchFilter($item, $subFilter)) {
-                        return true;
-                    }
-                }
-                return false;
-            case All::getOperator():
-                foreach ($arguments[0] as $subFilter) {
-                    if (!$this->matchFilter($item, $subFilter)) {
-                        return false;
-                    }
-                }
-                return true;
-            case Equals::getOperator():
-                [$field, $value] = $arguments;
-                return $item[$field] == $value;
-            case GreaterThan::getOperator():
-                [$field, $value] = $arguments;
-                return $item[$field] > $value;
-            case GreaterThanOrEqual::getOperator():
-                [$field, $value] = $arguments;
-                return $item[$field] >= $value;
-            case LessThan::getOperator():
-                [$field, $value] = $arguments;
-                return $item[$field] < $value;
-            case LessThanOrEqual::getOperator():
-                [$field, $value] = $arguments;
-                return $item[$field] <= $value;
-            case In::getOperator():
-                [$field, $values] = $arguments;
-                return in_array($item[$field], $values, false);
-            case Like::getOperator():
-                [$field, $value] = $arguments;
-                return stripos($item[$field], $value) !== false;
-            default:
-                throw new \RuntimeException("Operation \"$operation\" is not supported");
-        }
+        $filterProcessor = $this->getFilterProcessor();
+        /* @var $filterProcessor PhpVariableFilterProcessor */
+        return $filterProcessor->execute($item, $filter);
```

Comment:
To indicate that it's not building a query we can rename method:

```suggestion
        return $filterProcessor->match($item, $filter);
```

Reply:
renamed